index.html

```diff
@@ -40,5 +61,361 @@ <h2>
+                    <ol>
+                      <li>
+                        <p>If no sources of type <var>T</var> are available,
+                        <a>reject</a> <var>p</var> with a new
+                        {{DOMException}} object whose
+                        {{DOMException/name}} attribute has the value
+                        {{NotFoundError}}.</p>
+                      </li>
+                      <li>
+                        <p>Read the current [= permission state=] for obtaining
+                        sources of type <var>T</var> in the current browsing
+                        context. If the permission state is {{PermissionState/"denied"}}, jump to
+                        the step labeled <em>PermissionFailure</em> below.</p>
+                      </li>
+                    </ol>
+                  </li>
+                  <li>
+                    <p>Optionally, e.g., based on a previously-established
+                    user preference, for security reasons, or due to platform
+                    limitations, jump to the step labeled <em>Permission Failure</em> below.</p>
+                  </li>
+                  <li>
+                    <p>[=Request permission to use=]
+                    viewport capture, for a {{PermissionDescriptor}} with its
+                    {{PermissionDescriptor/name}} set to `"viewport-capture"`, resulting in
+                    a set of provided media.</p>
+                    <p>The provided media MUST include precisely one video track, which
+                    MUST be a live-capture of the
+                    <a data-cite="SCREEN-CAPTURE#dfn-browser">browser</a>
+                    <a data-cite="SCREEN-CAPTURE#dfn-display-surface">display surface</a>
+                    of the [=relevant settings object=]'s
+                    [=environment settings object/responsible document=]'s
+                    [=top-level browsing context=]'s
+                    <a data-cite="HTML#viewport">viewport</a>.</p>
+                    <p>The provided media MUST include at most one audio track, which, if
+                    provided, MUST be the combined audio produced by the sum of
+                    documents that consist of the
+                    [=relevant settings object=]'s
+                    [=environment settings object/responsible document=]'s
+                    [=top-level browsing context=]'s [=active document=], and
+                    all [=active document=]s in [=nested browsing context=]s of
+                    the [=relevant settings object=]'s
+                    [=environment settings object/responsible document=]'s
+                    [=top-level browsing context=]. This audio track
+                    MUST NOT be included if audio was not specified in
+                    <var>requestedMediaTypes</var>, or if it was specified as
+                    <code>false</code>.</p>
+                    <p>The source of a {{MediaStreamTrack}} MUST NOT change.</p>
```

Comment:
### 
```suggestion
```
redundant with existing getusermedia requirement

Reply:
> redundant with existing getusermedia requirement

@dontcallmedom where is that requirement?

If you mean this sentence: _"The source of the MediaStreamTrack MUST NOT change."_, then that's step 9.5.7. of the [getUserMedia](https://w3c.github.io/mediacapture-main/getusermedia.html#dfn-same-permission) algorithm, which is not called here.

I don't think there is a general requirement, or https://github.com/w3c/mediacapture-screen-share/pull/170 would be violating it.

Reply:
you're right, closing this thread then :)